vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -119,212 +117,139 @@ class OidcSiopWallet(
-            val params = AuthenticationRequestParameters.deserialize(jws.payload.decodeToString()).getOrElse {
-                return null
-                    .apply { Napier.w("parseRequestObjectJws: Deserialization failed", it) }
-            }
-            if (requestObjectJwsVerifier.invoke(jws, params))
-                AuthenticationRequestParametersFrom.JwsSigned(jws, params)
-            else null
-                .also { Napier.w("parseRequestObjectJws: Signature not verified for $jws") }
-        }
-    }
+    suspend fun parseAuthenticationRequestParameters(input: String) =
+        AuthenticationRequestParser.createWithDefaults(
+            remoteResourceRetriever = remoteResourceRetriever,
+            requestObjectJwsVerifier = requestObjectJwsVerifier,
+        ).parseAuthenticationRequestParameters(input)
 
     /**
      * Pass in the deserialized [AuthenticationRequestParameters], which were either encoded as query params,
      * or JSON serialized as a JWT Request Object.
      */
     suspend fun createAuthnResponse(
-        request: AuthenticationRequestParametersFrom<*>
+        request: AuthenticationRequestParametersFrom<*>,
     ): KmmResult<AuthenticationResponseResult> = catching {
         val response = createAuthnResponseParams(request).getOrThrow()
-        if (request.parameters.responseType == null
-            || (!request.parameters.responseType.contains(ID_TOKEN)
-                    && !request.parameters.responseType.contains(VP_TOKEN))
-        ) {
-            Napier.w("createAuthnResponse: Unknown response_type ${request.parameters.responseType}")
-            throw OAuth2Exception(Errors.INVALID_REQUEST)
-        }
-        when (request.parameters.responseMode) {
-            DIRECT_POST -> authnResponseDirectPost(request, response)
-            DIRECT_POST_JWT -> authnResponseDirectPostJwt(request, response)
-            QUERY -> authnResponseQuery(request, response)
-            FRAGMENT, null -> authnResponseFragment(request, response)
-            is OTHER -> TODO()
-        }
+        AuthenticationResponseFactory(jwsService).createAuthenticationResponse(
+            request,
+            response,
+        )
     }
 
-    private fun authnResponseDirectPost(
-        request: AuthenticationRequestParametersFrom<*>,
-        response: AuthenticationResponse
-    ): AuthenticationResponseResult.Post {
-        val url = request.parameters.responseUrl
-            ?: request.parameters.redirectUrl
-            ?: throw OAuth2Exception(Errors.INVALID_REQUEST)
-        return AuthenticationResponseResult.Post(url, response.params.encodeToParameters())
+    /**
+     * Creates the authentication response from the RP's [params]
+     */
+    suspend fun createAuthnResponseParams(
+        params: AuthenticationRequestParametersFrom<*>
+    ): KmmResult<AuthenticationResponse> = catching {
+        val preparationState = startAuthorizationResponsePreparation(params).getOrThrow()
```

Comment:
maybe inline this variable?

Reply:
fixed in f2a300dabe9855916935f35daf1408b409f71cc8